labkey/experiment.py

```diff
@@ -218,6 +218,55 @@ def save_batches(
     return None
 
 
+def lineage(
+    server_context: ServerContext,
+    lsids: List[str],
+    children: bool = None,
+    container_path: str = None,
+    cpas_type: str = None,
+    exp_type: str = None,
+    depth: int = None,
+    parents: bool = None,
+    run_protocol_lsid: str = None,
+):
+    """
+    :param server_context: A LabKey server context. See utils.create_server_context.
+    :param lsids: Array of LSIDs for the seed ExpData, ExpMaterials, or ExpRun
+    :param children: Include children in the lineage response. Defaults to true.
+    :param container_path: labkey container path if not already set in context
+    :param cpas_type: Optional LSID of a SampleSet or DataClass to filter the response. Defaults to include all.
+    :param exp_type: Optional experiment type to filter response -- either "Data", "Material", or "ExperimentRun".
+    Defaults to include all.
+    :param depth: An optional depth argument. Defaults to include all.
+    :param parents: Include parents in the lineage response. Defaults to true.
+    :param run_protocol_lsid: Optional Exp Run Protocol Lsid to filter response. Defaults to include all.
+    """
+    lineage_url = server_context.build_url(
+        "experiment", "lineage.api", container_path=container_path
+    )
+    payload = {"lsids": lsids}
+
+    if children is not None:
+        payload["children"] = children
+
+    if cpas_type is not None:
+        payload["cpasType"] = cpas_type
+
+    if exp_type is not None:
+        payload["expType"] = exp_type
+
+    if depth is not None:
+        payload["depth"] = depth
+
+    if parents is not None:
+        payload["parents"] = parents
+
+    if run_protocol_lsid is not None:
+        payload["runProtocolLsid"] = run_protocol_lsid
+
+    return server_context.make_request(lineage_url, payload=payload, method="GET")
```

Comment:
I'm not sure what we support on the server side, but any change the list of lsids array could be long enough where we'd want to do a POST here instead?

Reply:
I changed it to POST and it works the same, so I think I'll keep it that way to prevent any potential URL length issues.